lib/stat_tracker.rb

```diff
@@ -182,71 +182,79 @@ def lowest_scoring_home_team
   end
 
   # SEASON STATISTICS
+  # season stats helper methods -------------------
   def season_games(season)
     games.find_all { |game| game.season == season }
   end
+  #-----------------------------
 
   def winningest_coach(season)
-    #season_games_by_id returns an array of just season game ids
     season_game_ids = season_games(season).map do |game|
       game.game_id
     end
 
-    #then find all games in game_teams from the above season
     season_game_teams = game_teams.find_all do |game|
       season_game_ids.include?(game.game_id)
     end
 
-    # filter season games by wins
-    season_wins = season_game_teams.find_all { |game| game.result == "WIN" }
+    game_teams_by_coach = season_game_teams.group_by do |game_team|
+      game_team.head_coach
+    end
+
+    coach_games_and_wins = Hash.new { |h,k| h[k] = Hash.new(0) }
 
-    # returns an array of coach name for each win
-    coach_wins = season_wins.map { |game| game.head_coach }
+    game_teams_by_coach.each do |coach, game_teams|
+      coach_games_and_wins[coach][:games] = game_teams.count
+      coach_games_and_wins[coach][:wins] = game_teams.find_all { |game| game.result == "WIN"}.count
+    end
 
-    # creates a hash of number of season games won by coach
-    wins_by_coach = coach_wins.inject(Hash.new(0)) do |wins, coach|
-       wins[coach] += 1; wins
-     end
+    coach_win_percentage = {}
+    coach_games_and_wins.each do |coach, stats|
+      coach_win_percentage[coach] = stats[:wins].fdiv(stats[:games])
+    end
 
-    #return the winningest head_coach name as a string
-    coach_wins.max_by { |coach| wins_by_coach[coach] }
+    coach_win_percentage.max_by { |coach, record| record }[0]
   end
 
   def worst_coach(season)
     season_game_ids = season_games(season).map do |game|
       game.game_id
     end
 
-    season_games = game_teams.find_all do |game|
-    season_game_ids.include?(game.game_id)
+    season_game_teams = game_teams.find_all do |game|
+      season_game_ids.include?(game.game_id)
     end
 
-    season_losses = season_games.find_all do |game|
-    game.result == "LOSS"
+    game_teams_by_coach = season_game_teams.group_by do |game_team|
+      game_team.head_coach
     end
 
-    coach_losses = season_losses.map do |game|
-    game.head_coach
+    coach_games_and_wins = Hash.new { |h,k| h[k] = Hash.new(0) }
```

Comment:
@sciencefixion just an fyi - when you're setting your default value in a Hash to 0, you don't have to use the full notation! You can just write `Hash.new(0)` and that takes care of it :)